addons/mass_mailing/static/src/js/wysiwyg.js

```diff
@@ -27,7 +27,9 @@ const MassMailingWysiwyg = Wysiwyg.extend({
         const initialDropZone = this.$editable[0].querySelector('.o_mail_wrapper_td');
         const parsedHtml = new DOMParser().parseFromString(currentValue, "text/html");
         if (initialDropZone && !parsedHtml.querySelector('.o_mail_wrapper_td')) {
-            initialDropZone.replaceChildren(currentValue);
+            initialDropZone.innerHTML = currentValue;
+            this.odooEditor.sanitize();
+            this.odooEditor.historyStep(true);
```

Comment:
IMO this is not the right fix because it is possible to edit the html in the codeview while not removing the `o_mail_wrapper_td` element and you still wouldn't have the html sanitized and the `historyStep`. I'm not sure we want this code executed all the time outside of the `if` though as sanitizing is probably not needed in most cases, nor is the `historyStep`. I think a better place to have the fix would be right after `setValue` is called when exiting the codeview.

Reply:
@dmo-odoo I did added value to `innerHTML` because when we selectAll in codeview then `o_mail_wrapper_td` also got removed and in template preview we saw HTML instead of template.To solve this part we need `innerHTML` instead of `replaceChildren`.

What i understand from your comment is that we need to `sanitized` html  and `historyStep` but outside of `setValue`.So IMO can we put sanitized and `historyStep` in `_toggleCodeView` right after `setValue` called.

Reply:
I just tested the `replaceChildren` thing and you're right for `innerHTML`.

Yes, `_toggleCodeView` of field html seems like a better place.

Could you make it into two different commits please ? Because ultimately it's two different issues on their own, I think.